hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionName.java

```diff
@@ -20,4 +22,15 @@ public String getName() {
     public TopicName getTopicName() {
         return topicName;
     }
+
+    public static SubscriptionName fromString(String string) {
+        String[] tokens = string.split("\\$");
+        checkArgument(tokens.length > 1, "Incorrect string format. Expected 'topic$subscription'. Found:'%s'", string);
+        return new SubscriptionName(tokens[1], TopicName.fromQualifiedName(tokens[0]));
+    }
+
+    @Override
+    public String toString() {
+        return topicName.qualifiedName() + "$" + name;
```

Comment:
I tried using this format but it's an absolute nightmare to deserilize and has plenty of corner cases when group, topic or subscription starts (or ends) with an underscore. I managed to cover simple cases with `regexp="(?<!_)_(?!_)"` though...

The advantage of using something other than an underscore to separate topic from subscription is that we use an underscore to escape dots in zookeeper paths so parsing such a string is significantly easier. If you don't mind I would rather add few extra rules to topic and subscription validation to not contain dollar sign which should be pretty sensible.